lib/vdc/DCUGRID.cpp

```diff
@@ -0,0 +1,630 @@
+#include <vector>
+#include <algorithm>
+#include <map>
+#include <iostream>
+#include <cmath>
+
+#include <vapor/GeoUtil.h>
+#include <vapor/DCUtils.h>
+#include <vapor/DCUGRID.h>
+#include <vapor/VAssert.h>
+
+using namespace VAPoR;
+using namespace std;
+
+namespace {
+
+// Attribute names and values for "dummy" UGRID mesh variable
+//
+const string cfRoleAttName = "cf_role";
+const string cfRoleValueName = "mesh_topology";
+const string topologyAttName = "topology_dimension";
+const string nodeCoordinatesAttName = "node_coordinates";
+const string faceNodeConnectivityAttName = "face_node_connectivity";
+const string faceDimensionAttName = "face_dimension";
+const string edgeNodeConnectivityAttName = "edge_node_connectivity";
+const string edgeDimensionAttName = "edge_dimension";
+const string faceEdgeConnectivityAttName = "face_edge_connectivity";
+const string faceFaceConnectivityAttName = "face_face_connectivity";
+const string edgeFaceConnectivityAttName = "edge_face_connectivity";
+const string boundaryNodeConnectivityAttName = "boundary_node_connectivity";
+const string faceCoordinatesAttName = "face_coordinates";
+const string edgeCoordinatesAttName = "edge_coordinates";
+
+// Attribute names and values for data variables
+//
+const string meshAttName = "mesh";
+const string locationAttName = "location";
+
+
+template<class T> int getVar(NetCDFCFCollection *ncdfc, size_t ts, string varname, T *buf)
+{
+    int fd = ncdfc->OpenRead(ts, varname);
+    if (fd < 0) return (fd);
+
+    int rc = ncdfc->Read(buf, fd);
+    if (rc < 0) return (fd);
+
+    return (ncdfc->Close(fd));
+}
+
+// Read a netcdf attribute of type int
+//
+void getAttInt(NetCDFCFCollection *ncdfc, string varName, string attName, int &v)
+{
+    v = 0;
+    vector<long> values;
+    ncdfc->GetAtt(varName, attName, values);
+    if (values.size()) { v = values[0]; }
+}
+
+// Read a netcdf attribute of type string
+//
+void getAttString(NetCDFCFCollection *ncdfc, string varName, string attName, string &v)
+{
+    v = "";
+    ncdfc->GetAtt(varName, attName, v);
+}
```

Comment:
It's considered easier to read code if your function name is "getSomething" and your function return type is "something." These two functions seem perfectly fit in that pattern.

Reply:
done